packages/runtime-core/src/utils/preload.ts

```diff
@@ -70,6 +70,7 @@ export function preloadAssets(
   assets: PreloadAssets,
   // It is used to distinguish preload from load remote parallel loading
   useLinkPreload = true,
+  root: HTMLElement = document.head,
```

Comment:
A fatal flaw with the current approach is that when you load something (or preload something) it gets cached by id in a global. Subsequent calls to retrieve assets no longer returns those assets. So ultimately, Federation only supports a single root per global. 😞 To fix this I think we'd need to make sure that this code:

https://github.com/iammerrick/module-federation-core/blob/7d26e1325c4520534ee976d131cc6a8e9391c8b5/packages/runtime-core/src/plugins/generate-preload-assets.ts#L248

Somehow respects caches on whether it's been loaded for a specific root. Not assuming a global loading approach.